R/utils_render_latex.R

```diff
@@ -314,3 +333,181 @@ create_source_note_component_l <- function(data) {
 
   source_note_component
 }
+
+# Function to build a vector of `group` rows in the table body
+create_group_rows <- function(n_rows,
+                              groups_rows_df) {
+
+  unname(
+    unlist(
+      lapply(
+        seq(n_rows),
+        FUN = function(x) {
+
+          if (!(x %in% groups_rows_df$row_start)) {
+            return("")
+          }
+
+          latex_group_row(
+            group_name = groups_rows_df[
+              which(groups_rows_df$row_start %in% x), "group_label"][[1]],
+            top_border = x != 1,
+            bottom_border = x != n_rows
+          )
+        }
+      )
+    )
+  )
+}
+
+# Function to build a vector of `body` rows
+create_body_rows <- function(n_rows,
+                             row_splits_body) {
+
+  unname(
+    unlist(
+      lapply(
+        seq_len(n_rows),
+        FUN = function(x) {
+          latex_body_row(content = row_splits_body[[x]], type = "row")
+        }
+      )
+    )
+  )
+}
+
+# Function to build a vector of `summary` rows in the table body
+create_summary_rows <- function(n_rows,
+                                n_cols,
+                                boxh,
+                                list_of_summaries,
+                                groups_rows_df,
+                                stub_available,
+                                summaries_present) {
```

Comment:
If `!stub_available || !summaries_present`, just return `rep_len("", n_rows)`

Reply:
Created the early exit by checking length of `list_of_summaries` and removing `stub_available` and `summaries_present` (both redundant).